client/my-sites/earn/components/commission-fees.tsx

```diff
@@ -7,39 +6,56 @@ type CommissionFeesProps = {
 	className?: string;
 	commission: number | null;
 	iconSize?: number;
+	siteSlug?: string | null;
 };
 
-const CommissionFees = ( { className, commission, iconSize = 16 }: CommissionFeesProps ) => {
+const CommissionFees = ( {
+	className,
+	commission,
+	iconSize = 16,
+	siteSlug,
+}: CommissionFeesProps ) => {
 	const translate = useTranslate();
 
-	return commission !== null ? (
-		<span className={ className }>
-			{ preventWidows(
-				translate(
-					'On your current plan, WordPress.com charges {{em}}%(commission)s{{/em}}.{{br/}} Additionally, Stripe charges are typically %(stripe)s. {{a}}Learn more{{/a}}',
-					{
-						args: {
-							commission: '' + parseFloat( commission ) * 100 + '%',
-							stripe: '2.9%+30c',
-						},
-						components: {
-							em: <em />,
-							br: <br />,
-							a: (
-								<ExternalLink
-									href={ localizeUrl(
-										'https://wordpress.com/support/wordpress-editor/blocks/payments/#related-fees'
-									) }
-									icon={ true }
-									iconSize={ iconSize }
-								/>
-							),
-						},
-					}
-				)
-			) }
-		</span>
-	) : null;
+	if ( commission === null ) {
+		return null;
+	}
+
+	const commissionFee = commission * 100;
+	const StripeFeesLink = (
+		<ExternalLink href="https://stripe.com/pricing" icon={ true } iconSize={ iconSize } />
+	);
+	const plansLink = `/plans/${ siteSlug }`;
+	let commissionFeesText;
+
+	if ( commission === 0 || ! siteSlug ) {
+		commissionFeesText = translate(
+			'With your current plan, the transaction fee for payments is %(commissionFee)d% (+ <StripeFeesLink>Stripe fees</StripeFeesLink>).',
+			{
+				args: {
+					commissionFee,
+				},
+				components: {
+					StripeFeesLink,
+				},
+			}
+		);
+	} else {
+		commissionFeesText = translate(
+			'With your current plan, the transaction fee for payments is %(commissionFee)d% (+ <StripeFeesLink>Stripe fees</StripeFeesLink>). <UpgradeLink>Upgrade to lower it.</UpgradeLink>',
```

Comment:
Any difference to do `translate( "With your current plan..." ) + translate( "Upgrade to lower it." )` or is it better to keep these together and translate first part twice, like I've done in above code?

Reply:
I don't think there's any significant difference between the two approaches in terms of translatability, because the first sentence should get picked by the translation memory and in practice it wouldn't need to be translated twice.

However, given that the entire second sentence is wrapped in a component, I think there might be a little performance benefit if you split them and wrap just the text with the translate call, in order to avoid the component interpolation, e.g.:
```js
<a href={ plansLink }>{ translate( 'Upgrade to lower it.' ) }</a>
```

Reply:
Nice, thanks! I also like how it simplifies code a bit not needing big if/else.

I was caught by "it" word in the sentence and was thinking it can be helpful to have sentences combined for easier translation in some languages.

Reply:
> I was caught by "it" word in the sentence and was thinking it can be helpful to have sentences combined for easier translation in some languages.

That's a good point - a translator comment should definitely be added here to provide more context about the string. Maybe something like `'Upgrade' refers to upgrading the current plan, 'lower it' refers to lowering the transaction fees.`

Reply:
@yuliyan can you take a look how I've now implemented the PR?

Reply:
@simison, it looks good to me. I just left one small comment regarding the use of context instead of translator comment.